cmd/manager/profileparser.go

```diff
@@ -142,6 +177,37 @@ func runProfileParser(cmd *cobra.Command, args []string) {
 
 	err = profileparser.ParseBundle(contentDom, pb, pcfg)
 
+	if err == nil {
+		// associate profile names with rules
+		profileRuleMap, err := fetchProfileRuleMap(pcfg)
+		if err != nil {
+			cmdLog.Error(err, "Couldn't fetch profile rule map")
+			os.Exit(1)
+		}
+
+		ruleList := cmpv1alpha1.RuleList{}
+		err = pcfg.Client.List(context.TODO(), &ruleList, client.InNamespace(pcfg.ProfileBundleKey.Namespace))
+		if err != nil {
+			cmdLog.Error(err, "Couldn't list rules")
+			os.Exit(1)
+		}
+
+		for _, rule := range ruleList.Items {
```

Comment:
This is iterating over all rules that exist in the data stream, including those that aren't in any profile.

It seems to me that we know what are the rules selected in any profile. When we iterate over all profiles in `fetchProfileRuleMap()`, and from each profile we can get to all the rules with `profile.Rules.`
We can iterate only on the rules selected in a profile.

Reply:
Interesting remark about having too many updates done.

Yes, that is what I had in mind.

Current implementation is iterating over all profiles once.
Then, for each rule in the data stream (even those not selected) it is iterating over rules selected in a profile.
I think this is going O(n²).


During "fetch" phase it could already sort the rules into a map, with the value being a list of profiles.
Then, during "update" phase, it can iterate over the keys of the map, and directly use the value to update the rule.